Library/Homebrew/utils/shared_audits.rb

```diff
@@ -12,10 +12,15 @@ module SharedAudits
   def self.eol_data(product, cycle)
     @eol_data ||= T.let({}, T.nilable(T::Hash[String, T.untyped]))
     @eol_data["#{product}/#{cycle}"] ||= begin
-      result = Utils::Curl.curl_output("--location", "https://endoflife.date/api/#{product}/#{cycle}.json")
-      json = JSON.parse(result.stdout) if result.status.success?
-      json = nil if json&.dig("message")&.include?("Product not found")
-      json
+      result = Utils::Curl.curl_output("--location", "https://endoflife.date/api/v1/products/#{product}/releases/#{cycle}")
+
+      if result.status.success?
+        begin
+          JSON.parse(result.stdout)
+        rescue JSON::ParserError
+          nil
+        end
+      end
```

Comment:
This might be easier to read 
```suggestion
      return unless result.status.success?

      json = begin
        JSON.parse(result.stdout)
      rescue JSON::ParserError
        nil
      end
```

Reply:
That's what I had originally but unfortunately it doesn't pass `brew style` because RuboCop mistakenly thinks the return is occurring in a begin...end block.